wolfCLU/src/crypto/clu_crypto_setup.c

```diff
@@ -0,0 +1,358 @@
+                if (ret != 0) {
+                    printf("failed during conversion of IV, ret = %d\n",
+                           ret);
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return -1;
+                }
+                ivCheck = 1;
+            }
+            break;
+
+        case WOLFCLU_SIGN:
+            break;
+
+        case WOLFCLU_VERIFY:   /* Verify results, used with -iv and -key    */
+            /* using hexidecimal format */
+            inputHex = 1;
+            break;
+
+        case WOLFCLU_INFORM:   /* Certificate Stuff                         */
+        case WOLFCLU_OUTFORM:
+        case WOLFCLU_OUTPUT:
+        case WOLFCLU_NOOUT:
+        case WOLFCLU_TEXT_OUT:
+        case WOLFCLU_SILENT:
+        case WOLFCLU_PUBIN:
+        case WOLFCLU_PUBOUT:
+        case WOLFCLU_PUBKEY:
+
+
+            /* The cases above have their arguments converted to lower case */
+            if (optarg) convert_to_lower(optarg, (int)XSTRLEN(optarg));
+            /* The cases below won't have their argument's molested */
+
+        case WOLFCLU_INFILE:   /* File passed in by user                    */
+            in = optarg;
+            inCheck = 1;
+            break;
+
+        case WOLFCLU_OUTFILE:  /* Output file                               */
+            out = optarg;
+            outCheck = 1;
+            break;
+
+        case WOLFCLU_INKEY:
+            if (optarg == NULL) {
+                printf("no key passed in..\n");
+                wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                return FATAL_ERROR;
+            }
+
+            /* 2 characters = 1 byte. 1 byte = 8 bits
+             * number of characters / 2 = bytes
+             * bytes * 8 = bits
+             */
+            numBits = (int) (XSTRLEN(optarg) / 2 ) * 8;
+            /* Key for encryption */
+            if ((int)numBits != size) {
+                printf("Length of key provided was: %d.\n", numBits);
+                printf("Length of key expected was: %d.\n", size);
+                printf("Invalid Key. Must match algorithm key size.\n");
+                wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                return FATAL_ERROR;
+            }
+            else {
+                char keyString[strlen(optarg)];
+                XSTRNCPY(keyString, optarg, XSTRLEN(optarg));
+                ret = wolfCLU_hexToBin(keyString, &key, &numBits,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL);
+                if (ret != 0) {
+                    printf("failed during conversion of Key, ret = %d\n",
+                           ret);
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return -1;
+                }
+                keyCheck = 1;
+                keyType = 2;
```

Comment:
Same magic number comment as earlier. Prefer enum with names that mean something.